src/DataStructures/Tags.hpp

```diff
@@ -0,0 +1,25 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/Metafunctions.hpp"
+
+/// \cond
+class ModalVector;
+/// \endcond
+
+namespace Tags {
+/// Given the `NodalTag` swap the `Tensor::type` with a `ModalVector`.
```

Comment:
I have a mild concern about the potential misuse of this tag. For example, if `NodalTag` wraps a `Tensor<double>` or a `Tensor<CompletDataVector>`, then `Modal` will happily generate a valid tag ... but probably not the tag that was intended. In general we want our tags to validate types and avoid mistakes like this, so perhaps it makes sense to make this a little more "defensive".

I'm not sure if we should go to so far as to require from the compiler that `Tensor::type` be a `DataVector`, but perhaps the documentation should at least caution that the resulting Tag is (probably) only meaningful when `Tensor::type` is indeed a `DataVector`.

Reply:
I thought about this too. I think a `static_assert` here would be good.

Reply:
👍

Reply:
Pushed fixup :)

Reply:
Can you update this comment to reflect that the NodalTag is less generic now? I suggest "Wraps a `Tensor<ModalVector>` that has the same structure as the given `NodalTag`'s `Tensor<DataVector>`", or similar.